src/main/asciidoc/client-applications.adoc

```diff
@@ -363,6 +363,101 @@ include::{python-examples}/config_trust_example.py[tags=config-trust]
 ====
 
 
+[[driver-config-connection-pool-management]]
+=== Connection pool management
+
+The driver maintains a pool of connections. The pooled connections are reused by sessions and transaction to avoid overhead added by establishing new connections every time to run a query.
+The connections in the pool are created on demand. The connection pool always start with no pooled connection. New connections are created on demand by sessions and/or transactions. When sessions and/or transactions finished using connections, healthy ones are returned to the pool. These pooled connections are then reused by new sessions and transactions to execute queries.
+
+Three connection pool related configuration settings are exposed via driver configuration, namely `maxConnectionLifetime`, `maxConnectionPoolSize` and `connectionAcquisitionTimeout`.
```

Comment:
The name of the settings are following java parameter naming pattern, suggest to avoid java variables by using `maximum connection lifetime`, etc.